app/decorators/book_decorator.rb

```diff
@@ -0,0 +1,21 @@
+class BookDecorator < Draper::Decorator
+  decorates_association :authors
+
+  def self.collection_decorator_class
+    PaginatingDecorator
+  end
+
+  delegate_all
+
+  def authors_names
+    authors.map(&:full_name).join(', ')
+  end
+
+  def categories_names
+    categories.pluck(:name).join(', ')
+  end
+
+  def price
+    Money.new(price_cents, 'EUR')
```

Comment:
You have `Money.default_currency = Money::Currency.new('EUR')` so you don't need to pass currency here
```suggestion
    Money.new(price_cents)
```
Also you can use `monetize :price_cents` in Book model and remove this method from decorator